test/propagation_test.cpp

```diff
@@ -167,13 +167,31 @@ TEST_CASE("sampling behaviour") {
   std::shared_ptr<Tracer> tracer{new Tracer{tracer_options, buffer, getRealTime, getId, sampler}};
   ot::Tracer::InitGlobal(tracer);
 
+  // There's two ways we can set the sampling priority. Either directly using the method, or
+  // through a tag. Test both.
+  auto setSamplingPriority = GENERATE(
+      values<
+          std::function<OptionalSamplingPriority(Span*, std::unique_ptr<UserSamplingPriority>)>>({
+          [](Span* span, std::unique_ptr<UserSamplingPriority> p) {
+            return span->setSamplingPriority(std::move(p));
+          },
+          [](Span* span, std::unique_ptr<UserSamplingPriority> p) {
+            if (p != nullptr) {
+              span->SetTag("sampling.priority", static_cast<int>(*p));
+            } else {
+              span->SetTag("sampling.priority", "");
```

Comment:
Should it just be unset?

Reply:
That's what this does. So in addition to the spec of "0 = discard, anything else = keep" I'm adding a way to unset the sampling priority, by passing in an empty string.